pytest_container/pod.py

```diff
@@ -0,0 +1,217 @@
+"""Module for managing podman pods."""
+import contextlib
+from dataclasses import dataclass
+from dataclasses import field
+from pathlib import Path
+from subprocess import check_output
+from types import TracebackType
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from _pytest.mark import ParameterSet
+from filelock import FileLock
+
+from pytest_container.container import Container
+from pytest_container.container import ContainerData
+from pytest_container.container import ContainerLauncher
+from pytest_container.container import create_host_port_port_forward
+from pytest_container.container import DerivedContainer
+from pytest_container.container import PortForwarding
+from pytest_container.logging import _logger
+from pytest_container.runtime import get_selected_runtime
+from pytest_container.runtime import PodmanRuntime
+
+
+@dataclass
+class Pod:
+    """A pod is a collection of containers that share the same network and port
+    forwards. Currently only :command:`podman` supports creating pods.
+
+    **Caution**: port forwards of the individual containers are ignored and only
+    the port forwards in the pod class are taken into account!
+
+    """
+
+    #: containers belonging to the pod
+    containers: List[Union[DerivedContainer, Container]]
+
+    #: ports exposed by the pod
+    port_forwardings: List[PortForwarding] = field(default_factory=list)
+
+
+@dataclass(frozen=True)
+class PodData:
+    """Class that is returned by the ``pod`` and ``pod_per_test`` fixtures. It
+    contains all necessary information about the created pod and the containers
+    running inside it.
+
+    """
+
+    #: containers that are in the pod
+    containers: List[Union[DerivedContainer, Container]]
```

Comment:
Can we not have an instance of the Pod class here instead of explicitly redefining both `containers` and `forwarded_ports` attributes here?